samples/boards/pine64_pinetime/src/main.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2020 Stephane Dorre <stephane.dorre@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr.h>
+#include <drivers/gpio.h>
+
+static void button_pressed(struct device *dev,
+			   struct gpio_callback *cb, u32_t pins)
+{
+	printk("Button pressed at %" PRIu32 "\n", k_cycle_get_32());
+}
+
+static struct gpio_callback button_cb;
+
+int main(void)
+{
+	int ret = 0;
+	struct device *dev_led = NULL;
+	struct device *dev_btn = NULL;
+
+	dev_led = device_get_binding(DT_GPIO_LABEL(DT_ALIAS(led0), gpios));
+	if (!dev_led) {
+		return (-EOPNOTSUPP);
+	}
+
+	ret = gpio_pin_configure(dev_led, DT_GPIO_PIN(DT_ALIAS(led0), gpios),
+			   GPIO_OUTPUT |
+			   DT_GPIO_FLAGS(DT_ALIAS(led0), gpios));
+	if (ret != 0) {
+		printk("Error %d: failed to configure pin %d '%s'\n",
+			ret, DT_GPIO_PIN(DT_ALIAS(led0), gpios), DT_LABEL(DT_ALIAS(led0)));
+		return ret;
+	}
+
+	dev_btn = device_get_binding(DT_GPIO_LABEL(DT_ALIAS(sw0), gpios));
+	if (!dev_btn) {
+		return (-EOPNOTSUPP);
+	}
+
+	ret = gpio_pin_configure(dev_btn, DT_GPIO_PIN(DT_ALIAS(sw0), gpios),
+				 GPIO_INPUT | GPIO_INT_EDGE_BOTH |
+				 DT_GPIO_FLAGS(DT_ALIAS(sw0), gpios));
+	if (ret != 0) {
+		printk("Error %d: failed to configure pin %d '%s'\n",
+			ret, DT_GPIO_PIN(DT_ALIAS(sw0), gpios), DT_LABEL(DT_ALIAS(sw0)));
+		return ret;
+	}
+
+	gpio_init_callback(&button_cb, button_pressed,
+			   BIT(DT_GPIO_PIN(DT_ALIAS(sw0), gpios)));
+
+	gpio_add_callback(dev_btn, &button_cb);
+
+	ret = gpio_pin_configure(dev_btn, DT_GPIO_PIN(DT_ALIAS(sw1), gpios),
+				 GPIO_OUTPUT |
+				 DT_GPIO_FLAGS(DT_ALIAS(sw1), gpios));
+	if (ret != 0) {
+		printk("Error %d: failed to configure pin %d '%s'\n",
+			ret, DT_GPIO_PIN(DT_ALIAS(sw1), gpios), DT_LABEL(DT_ALIAS(sw1)));
+		return ret;
+	}
+
+	/* Pinetime trick : Enable button */
+	gpio_pin_set(dev_led, DT_GPIO_PIN(DT_ALIAS(sw1), gpios), 1u);
```

Comment:
`dev_led` here is a bit confusing, as the button state is set, and the button pin was configured using `dev_btn`.
But since there is only one GPIO peripheral in nRF52832, `dev_led` and `dev_btn` will always point to the same device (perhaps then it would be better to use `dev_gpio` instead of those two).
You can also use `GPIO_OUTPUT_ACTIVE` in the above call to `gpio_pin_configure()` and avoid the need of calling `gpio_pin_set()`.